libraries/mediaplayer/impl/src/main/kotlin/io/element/android/libraries/mediaplayer/impl/MediaPlayerImpl.kt

```diff
@@ -76,12 +89,16 @@ class MediaPlayerImpl @Inject constructor(
     private val scope = CoroutineScope(Job() + Dispatchers.Main)
     private var job: Job? = null
 
-    private val _state = MutableStateFlow(MediaPlayer.State(false, null, 0L, 0L))
+    private val _state = MutableStateFlow(MediaPlayer.State(MediaPlayer.PlayState.Stopped, null, 0L, 0L))
 
     override val state: StateFlow<MediaPlayer.State> = _state.asStateFlow()
 
-    override fun acquireControlAndPlay(uri: String, mediaId: String, mimeType: String) {
+    override fun setMedia(uri: String, mediaId: String, mimeType: String, playWhenReady: Boolean) {
+        player.playWhenReady = false
         player.clearMediaItems()
+        _state.update {
+            it.copy(playState = MediaPlayer.PlayState.Stopped)
```

Comment:
What is the difference between being Stopped and being Paused?
From the point of view of ExoPlayer there is no difference (we're still calling `prepare()` afterwards so the ExoPlayer will not be stopped, because stop means `STATE_IDLE`), so I feel we're introducing this `Stopped` state because of a need of `VoiceMessageComposerPlayer` rather than exposing an actual state of the ExoPlayer.

Reply:
The `Stopped` state models the state where the player has not yet been played or paused, the distinction being:
- in the `Stopped` state, the player should display the total duration and not display a cursor
- in the `Paused` state, the player should instead display the current player position and a cursor

I added this to the media player itself as I thought other parts of the app will likely want to share this behaviour. But agree that it might be a UI concern and could be a bit confusing for the model to be out of sync with the underlying media player.

I'll see if we can move this out into the presentation layer.